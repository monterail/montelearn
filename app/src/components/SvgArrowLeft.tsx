import React from "react";

const SvgArrowLeft = (props: any) => {
  return (
    <svg
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <path
        d="M23.25 12.497H0.75"
        stroke="#343541"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
      <path
        d="M4.5 16.247L0.75 12.497L4.5 8.74701"
        stroke="#343541"
        strokeWidth="1.5"
        strokeLinecap="round"
        strokeLinejoin="round"
      />
    </svg>
  );
};

export default SvgArrowLeft;
