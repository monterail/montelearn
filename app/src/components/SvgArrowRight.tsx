import React from "react";

const SvgArrowLeft = (props: any) => {
  return (
    <svg
      width="24"
      height="24"
      viewBox="0 0 24 24"
      fill="none"
      xmlns="http://www.w3.org/2000/svg"
      {...props}
    >
      <g clipPath="url(#clip0)">
        <path
          d="M0.75 11.503L23.25 11.503"
          stroke="#343541"
          strokeWidth="1.5"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
        <path
          d="M19.5 7.75299L23.25 11.503L19.5 15.253"
          stroke="#343541"
          strokeWidth="1.5"
          strokeLinecap="round"
          strokeLinejoin="round"
        />
      </g>
      <defs>
        <clipPath id="clip0">
          <rect x="24" y="24" width="24" height="24" transform="rotate(180 24 24)" fill="white" />
        </clipPath>
      </defs>
    </svg>
  );
};

export default SvgArrowLeft;
