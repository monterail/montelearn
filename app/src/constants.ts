export enum COOKIES {
  accessToken = "access_token",
  refreshToken = "refresh_token",
  firstName = "first_name",
  lastName = "last_name",
  isTeacher = "is_teacher",
  email = "email",
  pk = "pk",
}
