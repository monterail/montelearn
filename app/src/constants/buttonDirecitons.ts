export enum BUTTON_DIRECTIONS {
  LEFT = "left",
  RIGHT = "right",
}
