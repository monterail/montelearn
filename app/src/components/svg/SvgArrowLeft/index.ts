import SvgArrowLeft from "./SvgArrowLeft";

export default SvgArrowLeft;
