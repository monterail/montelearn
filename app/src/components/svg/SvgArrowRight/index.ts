import SvgArrowRight from "./SvgArrowRight";

export default SvgArrowRight;
