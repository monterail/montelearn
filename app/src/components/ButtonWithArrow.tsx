import { FunctionComponent } from "react";

import SvgArrowLeft from "@/components/svg/SvgArrowLeft";
import SvgArrowRight from "@/components/svg/SvgArrowRight";

import { BUTTON_DIRECTIONS } from "@/constants/buttonDirecitons";

type Props = {
  direction: BUTTON_DIRECTIONS;
  className?: string;
  withBorder?: boolean;
  onClick?: VoidFunction;
};

const ButtonWithArrow: FunctionComponent<Props> = ({
  direction,
  children,
  className = "",
  withBorder = true,
  onClick,
}) => {
  const directionClass = direction.charAt(0);
  const borderClasses = withBorder
    ? `px-6 border-2 border-black`
    : `pr-2 pl-4 sm:px-4 border-0 sm:border-2 border-black`;

  return (
    <button
      onClick={onClick}
      type="button"
      className={`
        flex py-2 whitespace-no-wrap hover:opacity-50 transition-opacity
        duration-200 bg-transparent rounded-full font-roboto-mono
        ${borderClasses} ${className}
      `}
    >
      {direction === BUTTON_DIRECTIONS.LEFT && <SvgArrowLeft />}
      <span className={`p${directionClass}-4 text-sm sm:text-base font-medium`}>{children}</span>
      {direction === BUTTON_DIRECTIONS.RIGHT && <SvgArrowRight />}
    </button>
  );
};

export default ButtonWithArrow;
