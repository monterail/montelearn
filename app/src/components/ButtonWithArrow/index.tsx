import ButtonWithArrow from "./ButtonWithArrow";

export default ButtonWithArrow;
