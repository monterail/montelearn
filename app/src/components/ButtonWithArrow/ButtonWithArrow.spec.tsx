import React from "react";
import { shallow } from "enzyme";

import { BUTTON_DIRECTIONS } from "@/constants/buttonDirecitons";
import SvgArrowLeft from "@/components/svg/SvgArrowLeft";
import SvgArrowRight from "@/components/svg/SvgArrowRight";
import ButtonWithArrow from "./ButtonWithArrow";

describe("ButtonWithArrow", () => {
  describe("is defined/matches snapshot", () => {
    it("should be defined", () => {
      expect(ButtonWithArrow).toBeDefined();
    });
    it("should render correctly", () => {
      const tree = shallow(<ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT} />);
      expect(tree).toMatchSnapshot();
    });
  });

  describe("direction", () => {
    it("should render with SvgArrowLeft", () => {
      const wrapper = shallow(<ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT} />);
      expect(
        wrapper
          .find("button")
          .children()
          .contains(<SvgArrowLeft />),
      ).toEqual(true);
    });
    it("should render with SvgArrowRight", () => {
      const wrapper = shallow(<ButtonWithArrow direction={BUTTON_DIRECTIONS.RIGHT} />);
      expect(
        wrapper
          .find("button")
          .children()
          .contains(<SvgArrowRight />),
      ).toEqual(true);
    });
  });

  describe("className", () => {
    const customClass = "custom-class";
    it("should render without custom class", () => {
      const wrapper = shallow(<ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT} />);
      expect(wrapper.find("button").hasClass(customClass)).toBe(false);
    });
    it("should render with custom class", () => {
      const wrapper = shallow(
        <ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT} className={customClass} />,
      );
      expect(wrapper.find("button").hasClass(customClass)).toBe(true);
    });
  });

  describe("withBorder", () => {
    it("should render with (default) border", () => {
      const wrapper = shallow(<ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT} />);
      expect(wrapper.find("button").hasClass("px-6 border-2 border-black")).toBe(true);
      expect(
        wrapper.find("button").hasClass("pr-2 pl-4 sm:px-4 border-0 sm:border-2 border-black"),
      ).toBe(false);
    });
    it("should render without border", () => {
      const wrapper = shallow(
        <ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT} withBorder={false} />,
      );
      expect(wrapper.find("button").hasClass("px-6 border-2 border-black")).toBe(false);
      expect(
        wrapper.find("button").hasClass("pr-2 pl-4 sm:px-4 border-0 sm:border-2 border-black"),
      ).toBe(true);
    });
  });

  describe("onClick", () => {
    const mockOnClick = jest.fn();
    it("should call mock function when button is clicked", () => {
      const wrapper = shallow(
        <ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT} onClick={mockOnClick} />,
      );
      wrapper.simulate("click");
      expect(mockOnClick).toHaveBeenCalled();
    });
  });

  describe("children", () => {
    const buttonText = "ButtonWithArrow text";
    it("should render ButtonWithArrow with custom text", () => {
      const wrapper = shallow(
        <ButtonWithArrow direction={BUTTON_DIRECTIONS.LEFT}>{buttonText}</ButtonWithArrow>,
      );
      expect(wrapper.find("button").text()).toMatch(buttonText);
    });
  });
});
