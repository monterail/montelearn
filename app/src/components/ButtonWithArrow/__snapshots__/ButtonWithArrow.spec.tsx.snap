// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonWithArrow is defined/matches snapshot should render correctly 1`] = `ShallowWrapper {}`;
